src/test/java/com/google/javascript/cl2dts/MultiFileTest.java

```diff
@@ -26,6 +30,18 @@ public void shouldPruneProvidesFromNonrootFile() throws Exception {
         .generatesDeclarations(false, expected);
   }
 
+  @Test
+  public void shouldResolveForwardDeclares() throws Exception {
+    String expected = DeclarationGeneratorTests.getTestFileText(input("forward.d.ts"));
+    ArrayList<File> roots = new ArrayList<>();
```

Comment:
because my bag of Java tricks is not that large (remember this is my first Java project :))